network/src/lib.rs

```diff
@@ -55,14 +55,20 @@ pub struct JustifiedBlockAnnounceValidator<B, P> {
 	phantom: PhantomData<B>,
 	polkadot_client: Arc<P>,
 	para_id: ParaId,
+	is_major_syncing: Box<dyn FnMut() -> bool + Send + Sync>,
```

Comment:
Why is that only a FnMut? Why not use the `SyncOracle`?

And please name it:

`polkadot_sync_oracle`.

Reply:
:thinking: fair point, you're totally right

Reply:
Can you check again? 6dec62e

Reply:
I think it is possible to not use Box if I expect the polkadot's "Service" (the network) instead of a SyncOracle

Reply:
LGTM